[CORE-9153] [CORE-9154] [CORE-9155] `rptest`: deflake `java_compression_test.py`

WillemKauf opened the pull request:
Another source of flakes in this test was the `VerifiableProducer`, [which will produce records without a key](https://github.com/a0x8o/kafka/blob/54eff6af115ee647f60129f2ce6a044cb17215d0/tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java#L300-L309) (i.e key is `null` for all records produced). This caused two failures:

1. The test was having trouble producing to the desired number of segments, as adjacent segment compaction seems to be able to keep up with the producer after sliding window compaction essentially reduces all of the segments to close to 0 bytes.
2. Because all records have the same `null` key, the consumer would not be able to consume the desired number of messages after all the records were compacted down to `1` visible record.

To fix this, specify `repeating_keys`, which will produce records with the key set cardinality specified. 

Also, the `min.cleanable.dirty.ratio` should have been set in commit https://github.com/redpanda-data/redpanda/commit/cba07b87f82bce719b33745391b0c02efd1c9e80 for unconditional compacting of the log per expectations of the test.

[Backporting to fix tests in previous versions.](https://github.com/redpanda-data/redpanda/pull/25157#issuecomment-2679744406) (the backports won't need the `min.cleanable.dirty.ratio` change, but they do need the other changes for deflaking.)

## Backports Required

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [X] v24.3.x
- [X] v24.2.x
- [X] v24.1.x

## Release Notes

* none

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--cac21767-5fa4-48fd-93fd-a0a9adb623a2-->
<!--TEST RESULTS-->
<!--5eebfb36-f6cc-4b25-b858-582f8a94851c-->
<!--TEST RESULTS-->
<!--9c664f1d-7da3-4a16-a8ea-2c4bebddc03b-->
<!--TEST RESULTS-->
<!--3818780b-9b6e-46cb-b811-a6db62f8093b-->
<!--TEST RESULTS-->
<!--b41051ef-f181-4cce-99be-cb4b9f7cd3e3-->
<!--TEST RESULTS-->
<!--81813b2e-3f88-4327-b5ea-e88f8c8fc55d-->
<!--TEST RESULTS-->
<!--2028fc14-b3c6-4eb1-ba10-5509df0e5140-->
<!--TEST RESULTS-->
<!--f1709e9e-6458-4b33-b3e8-b72533b8140f-->
<!--TEST RESULTS-->
<!--28ebba2c-76bd-43fb-bb24-7928efcc30dd-->
<!--TEST RESULTS-->
<!--9829c059-155c-4a16-9396-1a644c581ed1-->
<!--TEST RESULTS-->
<!--95056f7a-b144-4ad5-a549-7cd74325069d-->
<!--TEST RESULTS-->
<!--f73f53c9-8c84-4ebf-a3b1-42f8e0492dc1-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62485">build#62485</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| gtest_raft_rpunit.gtest_raft_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/62485#01954f19-b60d-44e3-aa3d-97e56a16bffc | FLAKY | 1/2 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62485#01954f74-8fa0-425e-807c-6f31d983505f | FLAKY | 1/2 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62485#01954f74-8fa0-425e-807c-6f31d983505f | FLAKY | 1/2 |
| rptest.tests.partition_movement_test.SIPartitionMovementTest.test_shadow_indexing.num_to_upgrade=0.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/62485#01954f77-6350-4759-afc5-34164311bb3c | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62502">build#62502</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62502#01955049-ed42-4993-a691-85da63893e2a | FLAKY | 1/3 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/62502#01955066-0dfd-4a29-82f8-e31ab2739ed8 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62508">build#62508</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62508#0195524a-9526-4648-a83b-7494cc84bc88 | FLAKY | 1/4 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62508#0195524a-9525-4acd-ab2b-2fd5082877b8 | FAIL | 0/20 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62508#0195524a-9526-4648-a83b-7494cc84bc88 | FLAKY | 1/2 |
| rptest.tests.partition_balancer_test.PartitionBalancerTest.test_rack_awareness | ducktape | https://buildkite.com/redpanda/redpanda/builds/62508#01955236-9b42-4a25-af7b-15d21834d092 | FLAKY | 1/2 |
| shard_placement_table_test_rpunit.shard_placement_table_test_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/62508#019551ee-cfc2-4166-80a5-219e43363124 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62510">build#62510</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:----------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.availability_test.AvailabilityTests.test_recovery_after_catastrophic_failure | ducktape | https://buildkite.com/redpanda/redpanda/builds/62510#019552fe-c4b4-4b1f-aa15-b830a7ad17d2 | FLAKY | 1/2 |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape | https://buildkite.com/redpanda/redpanda/builds/62510#019552fe-c4b5-4997-92ce-e03f23d34e51 | FLAKY | 1/3 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_fast_node_addition | ducktape | https://buildkite.com/redpanda/redpanda/builds/62510#01955301-d701-473c-934e-abe2029db44c | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/62510#01955301-d701-48b6-8033-ad2b3c2334bc | FLAKY | 1/3 |
| rptest.transactions.producers_api_test.ProducersAdminAPITest.test_producers_state_api_during_load | ducktape | https://buildkite.com/redpanda/redpanda/builds/62510#019552fe-c4b6-4441-9042-184d2b778698 | FLAKY | 1/2 |
| storage_e2e_single_thread_rpunit.storage_e2e_single_thread_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/62510#019552ba-2419-46ec-ab5d-9dcc128f38e0 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62513">build#62513</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b4ff-4490-9165-0ee7f8d98dd3 | FLAKY | 20/36 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b4ff-49be-b03b-d35a88462311 | FLAKY | 20/27 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b4ff-4e48-bb48-f4265116f3e3 | FLAKY | 20/28 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b500-43aa-80a2-73f31b0d27d6 | FLAKY | 20/32 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b500-463d-8fea-635b91d3755b | FLAKY | 20/36 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b500-4684-9bc5-b46f45ac4e07 | FLAKY | 20/34 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b500-46f9-b47a-8418e51b61e1 | FLAKY | 20/38 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b500-4e10-a6a0-6668e9f1c661 | FLAKY | 20/29 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b501-433e-aecf-1cd58afd05d1 | FLAKY | 20/32 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b501-488d-9258-e5d71e2ed53b | FLAKY | 20/38 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b50c-4f88-af00-456ca60f35fb | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b502-46ec-9c18-424fb08126c8 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b504-443a-be2d-93163c7fcef0 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b50e-45c5-99d3-b7394982a667 | FLAKY | 20/22 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b50f-4051-a446-4272d01904ad | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b50f-46a4-aafa-c410f3062221 | FLAKY | 20/23 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b50f-48d8-8172-950838a5fbe6 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b50f-4d4f-bfb4-7fa2d1c36c21 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b511-45b7-a2da-0107ef861f8e | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b511-46cf-b092-4990f06e4083 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b511-4d31-9df9-31a60ce31929 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b512-40a9-afa9-1a9e131ab5ec | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b512-4b51-82c4-e73b62fad510 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b512-4cd8-8b06-5794abf059e9 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b513-40ec-b857-e6b9b04f7e5e | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b507-47c4-bf3b-609f2aa1f37b | FLAKY | 20/24 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b507-4fe4-8c88-06f116972e4b | FLAKY | 20/24 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b508-43fc-838d-c051d0734dc8 | FLAKY | 20/23 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b508-44e0-ba72-0d996fda7efe | FLAKY | 20/23 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b508-4538-9803-dabe0dad6261 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b508-4eb5-bf90-3bb0ce3066aa | FLAKY | 20/24 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b508-4fb7-9829-fa3d1e89b630 | FLAKY | 20/24 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b509-40ed-a637-94a0ed927bbd | FLAKY | 20/27 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b509-491c-999d-74643cc8ea3a | FLAKY | 20/27 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b509-4ddc-97b6-d607ca9f1e22 | FLAKY | 20/22 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b513-42f4-9bec-fb65435db7f9 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b514-44a9-962e-c62d2eb74af6 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b514-4892-8c65-5f3bd8681b0a | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62513#0195538b-b515-4c88-b879-62a8aa0d3149 | FLAKY | 20/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62518">build#62518</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_storage_timing_stress_test.CloudStorageTimingStressTest.test_cloud_storage_with_partition_moves.cleanup_policy=compact.delete | ducktape | https://buildkite.com/redpanda/redpanda/builds/62518#019554b7-746b-4665-8faa-db46d3bfaa43 | FLAKY | 1/2 |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape | https://buildkite.com/redpanda/redpanda/builds/62518#0195549c-a03f-4019-a6bb-4f71ffd57c0c | FLAKY | 1/3 |
| rptest.tests.log_segment_ms_test.SegmentMsTest.test_segment_rolling.use_alter_cfg=False.server_cfg=None.topic_cfg=300000 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62518#0195549c-a03f-4d36-bdd9-61940e8e47fd | FLAKY | 1/2 |
| rptest.tests.mirror_maker_test.TestMirrorMakerService.test_simple_end_to_end.source_type=kafka | ducktape | https://buildkite.com/redpanda/redpanda/builds/62518#0195549c-a03f-4019-a6bb-4f71ffd57c0c | FLAKY | 1/2 |
| rptest.tests.partition_force_reconfiguration_test.PartitionForceReconfigurationTest.test_basic_reconfiguration.acks=-1.restart=False.controller_snapshots=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/62518#0195549c-a03d-4f37-a14b-aea7c6a0962c | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62520">build#62520</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff47-41d9-8c5f-f8e8968dd3a3 | FLAKY | 20/36 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff47-41fc-a9cf-e65e6ec3d3ea | FLAKY | 20/32 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff47-4aeb-a6ac-04905fd95ea8 | FLAKY | 20/30 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff47-4c6d-936f-32f03df45a69 | FLAKY | 20/29 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff47-4d72-bda1-288aab65e13e | FLAKY | 20/35 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff48-40cf-9623-f96de1740e44 | FLAKY | 20/34 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff48-4215-b5e8-022abfab206f | FLAKY | 20/33 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff48-46a4-84e8-75ad90193a4e | FLAKY | 20/31 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff48-4e90-9f98-a918d217994e | FLAKY | 20/39 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff48-4f59-a117-4f58507efa55 | FLAKY | 20/35 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff50-475f-8421-bc3fd20aac66 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff50-4ab3-9562-b1fda1636115 | FLAKY | 20/22 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff50-4acf-9c19-a3550a92191a | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4a-4291-ad84-cee3198ede2f | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4a-4f8a-9a62-bb2f833d97a5 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff52-4bf3-80cf-824801cc332a | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff53-45ff-ac34-4f722e702a3c | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff53-4bb9-bfd9-68ab3c515639 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff53-4d00-a3b9-b311f7194dd3 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff53-4d5e-aaf6-9d54997e87b4 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4c-4a32-97e6-761e86f4ac25 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff55-4147-91d1-233799cb314c | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff55-4311-85ff-94d0a2d2fe80 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff55-4446-b6e9-c47c9b46b8ed | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff55-45cb-a61b-1d684317e54e | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff56-46ae-bc40-8b9c64d9a901 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4d-43e5-8d58-7781f44ca750 | FLAKY | 20/25 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4d-4c48-9283-20e8e344b945 | FLAKY | 20/27 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4d-4d09-83df-d7eb9989fc30 | FLAKY | 20/27 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4e-4179-9a8a-1e71c41908cf | FLAKY | 20/24 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4e-41dc-a67b-c9db8627ef38 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4e-43f2-94eb-c7f8209169e2 | FLAKY | 20/23 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4e-4488-a0b8-06d2a68807d4 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4e-460f-aab2-6ada4f39d8e5 | FLAKY | 20/26 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4e-4d80-8994-635bfe9f5b2b | FLAKY | 20/26 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff4f-44ba-90af-76ba19b76678 | FLAKY | 20/23 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff57-4234-b3da-9974c661f342 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62520#01955687-ff58-4676-9719-608a639ff073 | FLAKY | 20/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62549">build#62549</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62549#01955ce8-58e8-4b47-bc85-34b8d4c71ee3 | FLAKY | 1/2 |
| rptest.tests.datalake.3rdparty_maintenance_test.Datalake3rdPartyMaintenanceTest.test_e2e_basic.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.TRINO.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62549#01955ceb-a0fa-4b51-9796-b2b2ca8bf8a0 | FLAKY | 1/2 |
| rptest.tests.node_pool_migration_test.NodePoolMigrationTest.test_migrating_redpanda_nodes_to_new_pool.balancing_mode=node_add.test_mode=TestMode.FAST_MOVES.cleanup_policy=compact.delete | ducktape | https://buildkite.com/redpanda/redpanda/builds/62549#01955ceb-a0fb-49b7-906c-e02e1ee599df | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62562">build#62562</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62562#01955df9-c016-4a26-8dbd-2b21e50f6cab | FLAKY | 1/2 |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape | https://buildkite.com/redpanda/redpanda/builds/62562#01955df9-c015-4e07-8f83-7a1cfd05a59a | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/62562#01955dfc-4311-4040-8e09-d209aabb7074 | FLAKY | 1/2 |
| rptest.tests.tiered_storage_pause_test.TestTieredStoragePause.test_safe_pause_resume.allow_gaps_topic_level=True.allow_gaps_cluster_level=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/62562#01955dfc-4312-4000-b2de-2a8e3e770676 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62611">build#62611</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699c-45fa-ad48-b6885cabe1a9 | FLAKY | 1/8 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699c-4b85-9276-ebb266f571ca | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562f2-f2fa-4718-9010-880695861e32 | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699e-4dd0-8665-883bcd72f573 | FLAKY | 1/2 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562f2-f2fb-4be1-958f-d149dc5588d3 | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699c-45fa-ad48-b6885cabe1a9 | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562f2-f2fb-41de-b0d5-f67ac014d17c | FLAKY | 1/8 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699c-4b85-9276-ebb266f571ca | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699d-4f9b-b2ac-680a7d812476 | FLAKY | 1/2 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699e-4dd0-8665-883bcd72f573 | FLAKY | 1/5 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699c-45fa-ad48-b6885cabe1a9 | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699c-4b85-9276-ebb266f571ca | FLAKY | 1/5 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562d8-699d-4f9b-b2ac-680a7d812476 | FLAKY | 1/3 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562f2-f2fa-4718-9010-880695861e32 | FLAKY | 1/5 |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/62611#019562f2-f2fb-4be1-958f-d149dc5588d3 | FLAKY | 1/2 |
| storage_e2e_single_thread_rpunit.storage_e2e_single_thread_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/62611#01956293-c50b-4441-b15f-861f9cfa5e86 | FLAKY | 1/2 |
| test_archiver_manager_smp_rpfixture.test_archiver_manager_smp_rpfixture | unit | https://buildkite.com/redpanda/redpanda/builds/62611#01956293-c508-4888-8d45-5e0d609e7d51 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62633">build#62633</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a2-41f5-8c87-aeefee148c96 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a3-43f0-a6d7-43c4f055837b | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a4-4458-8bde-e605f0a92539 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.GZIP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a5-4d74-8de5-99f6435bc8d1 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a5-4104-898d-eb956f0e66df | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a6-4120-9d88-0476ac5e04d0 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a6-45c2-917e-f91d16b47b80 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a6-4f40-ab8d-fb4e256d172e | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.LZ4 | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a7-4f91-b99d-f24d0b41d9bc | FLAKY | 20/22 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f69c-45bc-81a2-71b839a083d1 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a8-45f6-ace4-fa604514b0fe | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a9-42b5-a149-6c86911d6ff8 | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6a9-4892-98b3-45e09709f40d | FLAKY | 20/21 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.SNAPPY | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6aa-4a49-9ef7-5f1ba76bf015 | FLAKY | 20/22 |
| rptest.tests.compatibility.java_compression_test.JavaCompressionTest.test_upgrade_java_compression.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/62633#01956412-f6ac-46fc-b58c-2f600869c054 | FLAKY | 20/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62642">build#62642</a></summary>


| test_id                                                                                                                                                                                   | test_kind   | job_url                                                                                   | test_status   | passed   |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2f-48cb-b67f-ccac62b29052 | FLAKY         | 1/2      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e5-4c06-807c-687b4b19ca9a | FLAKY         | 1/5      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC                                | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2d-4830-8a0e-1c294b5351b9 | FLAKY         | 1/4      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_basic.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC                                | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e7-40bd-9a87-a24c3e747728 | FLAKY         | 1/3      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2e-4f72-980e-b599b397e861 | FLAKY         | 1/2      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e7-4e7e-8324-f0e9ffc3fe08 | FLAKY         | 1/6      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP             | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2f-48cb-b67f-ccac62b29052 | FLAKY         | 1/2      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP             | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e5-4c06-807c-687b4b19ca9a | FLAKY         | 1/6      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC               | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a30-4725-8753-7d851c729b87 | FLAKY         | 1/3      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC               | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e6-4ea6-9efd-a371be25d77d | FLAKY         | 1/5      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2d-4830-8a0e-1c294b5351b9 | FLAKY         | 1/3      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dlq.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC              | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2f-48cb-b67f-ccac62b29052 | FLAKY         | 1/5      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e6-4ea6-9efd-a371be25d77d | FLAKY         | 1/2      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2d-4830-8a0e-1c294b5351b9 | FLAKY         | 1/2      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e7-40bd-9a87-a24c3e747728 | FLAKY         | 1/4      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC     | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2e-4f72-980e-b599b397e861 | FLAKY         | 1/4      |
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_slow_tiered_storage_dupe_records.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC     | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e7-4e7e-8324-f0e9ffc3fe08 | FLAKY         | 1/3      |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact                                                               | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a30-4725-8753-7d851c729b87 | FLAKY         | 1/2      |
| rptest.tests.node_pool_migration_test.NodePoolMigrationTest.test_migrating_redpanda_nodes_to_new_pool.balancing_mode=node_add.test_mode=TestMode.FAST_MOVES.cleanup_policy=compact.delete | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956497-08e6-4ea6-9efd-a371be25d77d | FLAKY         | 1/2      |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move.replication_factor=3.unclean_abort=True.recovery=no_recovery.compacted=False       | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62642#01956493-3a2d-4830-8a0e-1c294b5351b9 | FLAKY         | 1/2      |
</details>
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/62508-->
<!--ddb93088-37d7-4a20-919e-b56d48cc529a-->
<!--title-->
## Retry command for [Build#62508](https://buildkite.com/redpanda/redpanda/builds/62508)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/compatibility/java_compression_test.py::JavaCompressionTest.test_java_compression@{"compression_type":"zstd"}
```
<!--/body-->

WillemKauf commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/compatibility/java_compression_test.py

WillemKauf commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/compatibility/java_compression_test.py

dotnwat reviewed (commented):
we should definitely deflake the test, but isn't one of the problems here that compacted topics should reject any produced records with null keys?

WillemKauf commented:
>we should definitely deflake the test, but isn't one of the problems here that compacted topics should reject any produced records with null keys?

Yup, good point.

https://cwiki.apache.org/confluence/display/KAFKA/KIP-135+%3A+Send+of+null+key+to+a+compacted+topic+should+throw+non-retriable+error+back+to+user

Opened https://redpandadata.atlassian.net/browse/CORE-9258.

WillemKauf committed:
`rptest`: use `repeated_keys` in `java_compression_test.py`

A source of flakes in this test was the `VerifiableProducer`, which will
produce records without a key (i.e key is `null` for all records produced).
This caused two failures:

1. The test was having trouble producing to the desired number of segments,
   as adjacent segment compaction seems to be able to keep up with the producer.
2. Because all records have the same `null` key, the consumer would not be able
   to consume the desired number of messages after all the records were compacted
   down to `1` visible record.

To fix this, specify `repeating_keys`, which will produce records with the key set
cardinality specified.

WillemKauf committed:
`rptest`: add extra logging to `wait_for_compacted_segments()`

In case of flakes, this log line can be good to diagnose errors.

WillemKauf committed:
`rptest`: set `min.cleanable.dirty.ratio` to `0` in `java_compression_test`

These tests expect unconditional compacting of the log.

WillemKauf committed:
`rptest`: account for active segment in `wait_for_compacted_segments`

Otherwise these tests are waiting a long time for a segment to roll,
or may timeout before that happens.

WillemKauf committed:
`rptest`: deflake `test_upgrade_java_compression`

This test expects that all segments will be compacted and re-written,
in order to recompress with the `snappy` compression fix post node restart.

However, this may not always occur for the test currently as it is written,
since some segments may be detected as already compacted post restart,
or not included in sliding window compaction due to already being cleanly
compacted.

To fix this, delete all the segment indices on disk to force a rebuild and
a self compaction of the segments.

Also, change the test to no longer wait on compaction, but on the consumer
succeeding instead.

dotnwat commented:
I think it makes sense to just keep 1 CORE-NNNN annotation in the PR title

dotnwat commented on tests/rptest/tests/compatibility/java_compression_test.py:
```diff
@@ -47,24 +47,28 @@ def produce_until_segment_count(self,
                                       topic=self.topic_spec.name,
-                                      compression_types=[compression_type])
+                                      compression_types=[compression_type],
+                                      repeating_keys=key_set_cardinality)
```
cool, just learned about this parameter

dotnwat commented on tests/rptest/tests/compatibility/java_compression_test.py:
```diff
@@ -47,24 +47,28 @@ def produce_until_segment_count(self,
         finally:
             producer.stop()
             producer.clean()
             producer.free()
 
-    def consume(self, num_messages=100, timeout_sec=120):
+    def consume(self, num_messages=10, timeout_sec=120):
```
seems like a random change. is there an explanation?

WillemKauf replied:
We don't need to see many messages in the consumer to assert on the test expectations.

dotnwat commented on tests/rptest/tests/compatibility/java_compression_test.py:
```diff
@@ -131,7 +131,7 @@ def test_java_compression(self, compression_type):
         expected_num_segments = 4
         self.produce_until_segment_count(
             expected_num_segments, compression_type=compression_type.value)
-        self.wait_for_compacted_segments(expected_num_segments)
+        self.wait_for_compacted_segments(expected_num_segments - 1)
```
makes sense. i guess the alternative would be to set a shorter value for apply_segment_ms so the active segment would roll?

WillemKauf replied:
Unneeded, I think.

WillemKauf commented:
Unfortunately this still flakes in the most recent build. Have to revisit tomorrow.

WillemKauf commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/compatibility/java_compression_test.py

WillemKauf commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/compatibility/java_compression_test.py

WillemKauf commented:
No flakes in [Build #62663](https://buildkite.com/redpanda/redpanda/builds/62663) :+1:

vbotbuildovich commented:
/backport v24.3.x

vbotbuildovich commented:
/backport v24.2.x

vbotbuildovich commented:
/backport v24.1.x

vbotbuildovich commented:
Failed to create a backport PR to v24.2.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-25227-v24.2.x-157 remotes/upstream/v24.2.x
git cherry-pick -x 8904f0d159 2d96bc4fa7 0916961d75 e9bbc721a6 24bb79d1a6
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/13734687328).

vbotbuildovich commented:
Failed to create a backport PR to v24.3.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-25227-v24.3.x-960 remotes/upstream/v24.3.x
git cherry-pick -x 8904f0d159 2d96bc4fa7 0916961d75 e9bbc721a6 24bb79d1a6
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/13734687450).

vbotbuildovich commented:
Failed to create a backport PR to v24.1.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-25227-v24.1.x-97 remotes/upstream/v24.1.x
git cherry-pick -x 8904f0d159 2d96bc4fa7 0916961d75 e9bbc721a6 24bb79d1a6
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/13734687511).
